### DISCUSSION TITLE
Downgrade Django

### DISCUSSION THREAD

**Discussion by MalikRumi** [python-poetry/poetry#8511] [General]
I recently discovered a conflict between Django 4.2 and my current project. I decided to downgrade to Django 4.0 until I can get that other issue fixed. However, I could not find a 'downgrade' command. I tried many variations on "poetry add Django@4.0" as shown in the [docs](https://python-poetry.org/docs/dependency-specification/#exact-requirements
) but I ended up getting an _upgrade_ to Django 4.2.6!  I tried just removing it, thinking i could then install 4.0, but that got me "No dependencies to install or update". Finally, I edited the .toml file directly, ran "poetry update" as requested, and presto, my problem, vanished - for now. So my question is simply this: Was this the right or only way to do this, or did I miss an easier way in the docs? Where can I find this better way? Thx.

**Comment by dimbleby** [comment]
I expect you have confused yourself somehow: `poetry add django@4.0` does exactly what you expected all along

**Comment by cankush625** [comment]
@MalikRumi If you specifically want to downgrade to Django 4.0, the correct way is to do that: poetry add django==4.0.0. This command is sufficient to downgrade your Django version from 4.2 to 4.0. The command that you mentioned should also do the same, but I haven't tested that.
Updating .toml directly and then running poetry update should also be fine and I don't see any harm in that. poetry add command also creates/updates the dependency entry in .toml file.
